packages/lesswrong/server/authenticationMiddlewares.ts

```diff
@@ -378,15 +378,34 @@ export const addAuthMiddlewares = (addConnectHandler) => {
   const auth0OAuthSecret = auth0OAuthSecretSetting.get()
   const auth0Domain = auth0DomainSetting.get()
   if (auth0ClientId && auth0OAuthSecret && auth0Domain) {
-    passport.use(new Auth0Strategy(
+    const auth0Strategy: any = new Auth0Strategy(
       {
         clientID: auth0ClientId,
         clientSecret: auth0OAuthSecret,
         domain: auth0Domain,
         callbackURL: combineUrls(getSiteUrl(), 'auth/auth0/callback')
       },
       createOAuthUserHandlerAuth0('services.auth0', profile => profile.id, userFromAuth0Profile)
-    ));
+    )
+    passport.use(auth0Strategy)
+
+    const userHandler = createOAuthUserHandler('services.auth0', profile => profile.id, userFromAuth0Profile)
+
+    addConnectHandler('/auth/useAccessToken', (req, res, next) => {
```

Comment:
I believe this is probably sound, but it is highly non-idiomatic. Can you rewrite this as a custom strategy, in the style of the cookie custom strategy? Or explain why that's a bad approach.

Reply:
That works, the only problem is that it needs access to the auth0Strategy object, so it can't be defined at the top-level like cookieAuthStrategy

Reply:
I think your solution is great